jwst/pipeline/calwebb_spec2.py

```diff
@@ -362,11 +362,11 @@ def process_exposure_product(
             x1d = self.extract_1d(x1d)
 
             # Possible that no fit was possible - if so, skip photom
-            if x1d is not None:
+            if (x1d is None) or (x1d.meta.cal_step.extract_1d == "SKIPPED"):
```

Comment:
Not sure this logic will really work as expected, at least the part that checks `x1d is None`, because up in line 361 `x1d` is populated from `resampled.copy()` and the logic in the extract_1d step that checks SOSS for error conditions (e.g. use of the F277W filter or FULL subarray) returns a copy of the input_model, as opposed to None. See, e.g., https://github.com/spacetelescope/jwst/blob/master/jwst/extract_1d/extract_1d_step.py#L432  So I don't think x1d will ever actually be None. Perhaps this also warrants fixing/updating things in the extract_1d step so that it does actually return None for those SOSS conditions.

Reply:
The check for None is related to [this line](https://github.com/spacetelescope/jwst/blob/95c8eb5ce89288f3ca056417c31c0bc527329f44/jwst/extract_1d/extract_1d_step.py#L471C3-L471C32), which covers the case that a fit could not be found by SOSS, not the checks for bad inputs that you mention above.

Reply:
Ah, OK. so both tests (the check for `x1d is None` and the cal_step being "SKIPPED") have a purpose, but in different circumstances. Sounds good.